### DISCUSSION TITLE
Registered ISO to cloudstack UI but it is not showing ready

### DISCUSSION THREAD

**Discussion by subincb30** [apache/cloudstack#9726] [Q&A]
I am trying to setup cloud stack environment for testing. But currently, I am facing issues with uploading ISO. It registered successfully but not getting uploaded. Also, I found the secondary storage VM in connecting state. 

Please find the attached images.
![Screenshot 2024-09-25 110341](https://github.com/user-attachments/assets/5ce418cc-1622-4d50-8cfd-16549ceca47e)
![Screenshot 2024-09-25 110414](https://github.com/user-attachments/assets/25c1ccaa-474f-4438-a0ed-f35c1d3449cc)

**Comment by DaanHoogland** [comment]
> Also, I found the secondary storage VM in connecting state.

This is definately the root cause of your issue. Ass the CPVM is up and connected, this is unexpected. It is most likely a networking issue, so try connection from the ssvm to internet/the MS/storage, and see if it has mounted the secondary storage.

**Comment by subincb30** [comment]
@DaanHoogland Since, this is for the testing purpose, I setup storage server in the same management server. As you suggested, I checked the internet connectivity and it is fine. Also, I am attaching the secondary storage mount status. 

---
![image](https://github.com/user-attachments/assets/f19c28fa-6a3c-4443-859c-06cd52b5f9cd)
---

**Reply by DaanHoogland** [reply]
how about other firewall rules? for ftp, http and https to the internet and port 8250 to the management server.

**Comment by subincb30** [comment]
Firewall is disabled. Please find the snippet below:

---
root@ubuntu:~# ufw status
Status: inactive
root@ubuntu:~# systemctl status apparmor
○ apparmor.service - Load AppArmor profiles
     Loaded: loaded (/lib/systemd/system/apparmor.service; disabled; vendor preset: enabled)
     Active: inactive (dead) since Wed 2024-09-25 09:46:13 UTC; 49s ago
       Docs: man:apparmor(7)
             https://gitlab.com/apparmor/apparmor/wikis/home/
   Main PID: 655 (code=exited, status=0/SUCCESS)
        CPU: 895us
---

@DaanHoogland

**Comment by kiranchavala** [comment]
@subincb30 

Could you please refer to this discussion if you are testing cloudstack 

https://github.com/apache/cloudstack/issues/7568

https://github.com/apache/cloudstack/issues/7568#issuecomment-1572043656

**Comment by bernardodemarco** [comment]
@subincb30, I already faced a similar issue and it was due to an incorrect secondary storage NFS setup.

When troubleshooting the situation, I first tested that the SSVM was able to access the internet and that the domain names were being correctly translated by the DNS servers (as you've already done on https://github.com/apache/cloudstack/discussions/9726#discussioncomment-10748419).

Then, I accessed the secondary storage mount point on the SSVM and performed some writing tests. Generally, the secondary storage has three nested directories: `snapshots`, `template` and `volumes`. I realized that I wasn't able to create new directories inside of them.     

So, I accessed the NFS storage server and removed any restrictions in the secondary storage directory permissions:

```bash
chmod -R 777 /mnt/sec-nfs-01/
```

Then, back at the secondary storage mount point on the SSVM, I checked that I was able to create nested directories. Given that, I accessed the CloudStack environment and registered successfully new ISOs and templates.

Could you validate your secondary storage NFS server settings and check if the SSVM has proper permissions to perform writing operations to the storage?

**Comment by subincb30** [comment]
@bernardodemarco this is my secondary storage permission:

---
drwxrwxrwx 3 nobody nogroup  4096 Sep 23 04:20 primary/
drwxrwxrwt 4 nobody nogroup  4096 Sep 23 04:19 secondary/
root@ubuntu#
root@ubuntu:/mnt/storage# pwd
/mnt/storage
---

I hope this is fine. Can you check and confirm?

**Reply by bernardodemarco** [reply]
Could you validate the permissions of the directories that are inside `/mnt/storage/secondary`? Besides that, as @rajujith
 mentioned on https://github.com/apache/cloudstack/discussions/9726#discussioncomment-10759429, it's great taking a look at the SSVM logs.

**Comment by rajujith** [comment]
@subincb30  you should be able to identify the cause of this issue if you can check the `/var/log/cloud.log` on the SSVM.

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
